### ISSUE TITLE
lib/charnames.t: blocker to strict-by-default

### ISSUE THREAD

**Issue by jkeenan** [atoomic/perl#227]
As of commit ac6b7cc in the alpha-dev-02-strict branch on Aug 18 2020, the file cited is failing. For each file, here is sample output from cd t; ./perl harness -v <file>; cd -:
```
Can't find Unicode property definition "name=/\ACJK UNIFIED IDEOGRAPH-59C3\z/" in regex; marked by <-- HERE in m/\p{name=/\ACJK UNIFIED IDEOGRAPH-59C3\z/} <-- HERE / at ../lib/charnames.t line 401.
../lib/charnames.t .. 
Dubious, test returned 255 (wstat 65280, 0xff00)
No subtests run 

Test Summary Report
-------------------
../lib/charnames.t (Wstat: 65280 Tests: 0 Failed: 0)
  Non-zero exit status: 255
  Parse errors: No plan found in TAP output
Files=1, Tests=0,  0 wallclock secs ( 0.02 usr  0.00 sys +  0.04 cusr  0.00 csys =  0.06 CPU)
Result: FAIL
```
@atoomic @toddr

Thank you very much.
Jim Keenan

**Comment by khwilliamson** [comment]
This problem stems from a change in mktables.  For some reason, you changed @code_points_ending_in_code_point to be in the package main::.  But the Name.pm file that this array is defined in is package _charnames.  The definition was written out without a package name so that it would take on the package given in the file.  

But now main is overriding that, and charnames can't find the array, so thinks it is empty, and regcomp.c is calling charnames to compile the pattern, and is being told there are no such names, so returns the error

**Comment by jkeenan** [comment]
> This problem stems from a change in mktables. For some reason, you changed @code_points_ending_in_code_point to be in the package main::. But the Name.pm file that this array is defined in is package _charnames. The definition was written out without a package name so that it would take on the package given in the file.
> 
> But now main is overriding that, and charnames can't find the array, so thinks it is empty, and regcomp.c is calling charnames to compile the pattern, and is being told there are no such names, so returns the error

Thanks.  This should be fixed by 7d5de5fe78.  Can you check that out?

Thank you very much.
Jim Keenan

**Comment by jkeenan** [comment]
> > This problem stems from a change in mktables. For some reason, you changed @code_points_ending_in_code_point to be in the package main::. But the Name.pm file that this array is defined in is package _charnames. The definition was written out without a package name so that it would take on the package given in the file.
> > But now main is overriding that, and charnames can't find the array, so thinks it is empty, and regcomp.c is calling charnames to compile the pattern, and is being told there are no such names, so returns the error
> 
> Thanks. This should be fixed by [7d5de5f](https://github.com/atoomic/perl/commit/7d5de5fe78046709402f5ec3bdc0fa3974b4979b). Can you check that out?
> 
> Thank you very much.
> Jim Keenan

@khwilliamson does my fix for this look okay?

Thanks.
jimk

**Comment by jkeenan** [comment]
Commit reviewed, okayed, closing.
